### PR TITLE
bug(axios): handle network error globally

### PR DESCRIPTION
**Pull request by openwell**
# Description

Describe change or new feature here.
This PR handles network error. By using axios interceptor as an HOC
## Checklist

Remove any items which are not applicable.

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works AND the tests pass

### CODE REVIEW

**Comment by openwell** [comment]
[Add farmer button does not handle network failure](https://trello.com/c/4BXTwcMe/162-add-farmer-button-does-not-handle-network-failure)

**Comment by openwell** [comment]
[Accept/Reject change buttons do not handle network failure](https://trello.com/c/kwjQMAHY/160-accept-reject-change-buttons-do-not-handle-network-failure)

**Review by Rexben001 — DISMISSED** [review]
- Check for redundant toast error messages in update farmers
- Stop the loader when there is a network error

File: src/components/pages/AddFarmer/AddFarmer.js [hunk 1]
```diff
@@ -148,12 +148,15 @@ const AddFarmer = ({ history, appStateShouldUpdate }) => {
...
-            toast.error(element.message);
-          });
+        if (err.response) {
+          if (Array.isArray(err.response.data.errors)) {
+            err.response.data.errors.forEach(element => {
+              toast.error(element.message);
+            });
+          } else if (err.response.data.errors.message) {
+            toast.error(err.response.data.errors.message);
+          }
```
Comment (quansenB) [inline src/components/pages/AddFarmer/AddFarmer.js]: could you add a 3. case in the case both conditional sattements don't execute?

**Comment by Rexben001** [comment]
I just finished QA 🔥

**Comment by Pav0l** [comment]
[User is logged out when changing his password](https://trello.com/c/AiQsIjOF/141-user-is-logged-out-when-changing-his-password)

File: src/components/pages/UpdateFarmer/UpdateFarmer.js [hunk 2]
```diff
@@ -200,13 +200,16 @@ const UpdateFarmer = ({ location, history, appStateShouldUpdate, user }) => {
...
- 
-        if(err.response.status === 403){
-        toast.warn(err.response.data.message)
+
+        if (Array.isArray(err.response.data.errors)) {
+          err.response.data.errors.forEach(element => {
+            toast.error(element.message);
+          });
+        } else if (err.response.data.errors.message) {
+          toast.error(err.response.data.errors.message);
```
Comment (Rexben001) [inline src/components/pages/UpdateFarmer/UpdateFarmer.js]: Nice

**Review by Pav0l — APPROVED** [review]

**Pull request merged**